### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by Piercel2022**
For this project, the following steps have been implemented
   1) Add the following validations:
        For the User model:
            Name must not be blank.
            PostsCounter must be an integer greater than or equal to zero.
        For the Post model:
            Title must not be blank.
            Title must not exceed 250 characters.
            CommentsCounter must be an integer greater than or equal to zero.
            LikesCounter must be an integer greater than or equal to zero.
   
2) Add unit specs for all of your models' methods and validations.

model-test

### CODE REVIEW

**Comment by DammyShittu** [comment]
Hi @Piercel2022,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### - All the headers listed in the readme template should be present. Kindly list them and ensure they are linked to sections on the readme
### - I cannot find the `spec` changes for this project. There is no `spec` folder in the root path pointing to the tests. Kindly inspect that (perhaps you did not push your spec files? 🤔).

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @Piercel2022, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
- [ ] [REQUIRED] Good job so far! You should have unit tests for the `User` ✔️, `Post` ✔️, `Comment` ❌, and `Like` ❌ model but I couldn't find unit tests for `Comment` and `Like`, please implement it 🙏🏼 
- `[HINT]`
```ruby
/* ------------ Comment--------------- */

require 'rails_helper'

RSpec.describe Comment, type: :model do
  pending "add some examples to (or delete) #{__FILE__}"
end

/* ------------ LIKE--------------- */

require 'rails_helper'

RSpec.describe Like, type: :model do
  pending "add some examples to (or delete) #{__FILE__}"
end

```


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AshakaE** [comment]
### Hi @Piercel2022  🙋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request ❌ 

You have submitted a project, that fails the linter checks

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    it 'Find post_id to be an integer' do
+      subject.post_id = 'caro'
+      expect(subject).to_not be_valid
+    end
+
+    it 'Find post_id to be an integer greater or equal to zero' do
+      subject.post_id = -2
+      expect(subject).to_not be_valid
+    end
+  end
```
Comment (anagudelogu) [inline spec/models/like_spec.rb:28]: - [ ] **[OPTIONAL]:** You did a great job following the previous reviewer's advice of writing unit tests for Likes. 🚀  However, I'm sorry that he didn't point out more details about what should be tested. in this case, you tested that a `like` should have an author and should be associated with a post but, there's a method `#likes_counter` in your model that is not tested, and, according to the requirements:

image

I strongly suggest you add a test to it! 🤓 🔧

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    it 'Find post_id to be an integer' do
+      subject.post_id = 'caro'
+      expect(subject).to_not be_valid
+    end
+
+    it 'Find post_id to be an integer greater or equal to zero' do
+      subject.post_id = -2
+      expect(subject).to_not be_valid
+    end
+  end
```
Comment (anagudelogu) [inline spec/models/comment_spec.rb:28]: - [ ] **[OPTIONAL]:** The same happens here, your Comment model has an `#update_coments_counter` method that is not tested. I strongly suggest you add a test to it! 🤓 🔧

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @Piercel2022, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**